src/plugins/skills/src/systems/slots.rs

```diff
@@ -1,36 +1,36 @@
 use crate::{
-	components::{slots::Slots, BoneName, Mounts, Slot, SlotBones},
-	items::slot_key::SlotKey,
+	components::{slots::Slots, BoneName, Mounts, Slot, SlotsDefinition},
+	items::{slot_key::SlotKey, Item},
 };
 use bevy::{
 	prelude::{BuildChildren, Children, Commands, Entity, HierarchyQueryExt, Name, Query},
 	scene::SceneBundle,
 };
-use common::traits::try_remove_from::TryRemoveFrom;
+use common::traits::{load_asset::Path, try_remove_from::TryRemoveFrom};
 
-pub(crate) fn add_item_slots(
+pub(crate) fn init_slots(
 	mut commands: Commands,
-	mut agent: Query<(Entity, &mut Slots, &mut SlotBones)>,
+	mut agent: Query<(Entity, &mut Slots<Path>, &mut SlotsDefinition)>,
 	children: Query<&Children>,
 	bones: Query<&Name>,
 ) {
 	for (agent, mut slots, mut slot_infos) in &mut agent {
-		let add_slot = |slot_info: (SlotKey, Mounts<&'static BoneName>)| {
-			let (key, mounts) = slot_info;
-			match find_bones(agent, &mounts, &children, &bones) {
-				Some((hand, forearm)) => {
-					slots.0.insert(
-						key,
-						Slot {
-							mounts: new_handles_on(hand, forearm, &mut commands),
-							item: None,
-						},
-					);
-					None
+		let add_slot =
+			|(key, (mounts, item)): (SlotKey, (Mounts<&'static BoneName>, Option<Item<Path>>))| {
+				match find_bones(agent, &mounts, &children, &bones) {
+					Some((hand, forearm)) => {
+						slots.0.insert(
+							key,
+							Slot {
+								mounts: new_handles_on(hand, forearm, &mut commands),
+								item,
+							},
+						);
+						None
+					}
+					None => Some((key, (mounts, item))),
 				}
-				None => Some((key, mounts)),
-			}
-		};
+			};
```

Comment:
Can we tidy this up?

Reply:
I tried my best, but Rest and lifetimes are a bit in the way of extracting stuff out.